hazelcast/proxy/map.py

```diff
@@ -1246,15 +1246,12 @@ def _on_destroy(self):
         super(MapFeatNearCache, self)._on_destroy()
 
     def _add_near_cache_invalidation_listener(self):
-        try:
-            codec = map_add_near_cache_invalidation_listener_codec
-            request = codec.encode_request(self.name, EntryEventType.INVALIDATION, self._is_smart)
-            self._invalidation_listener_id = self._register_listener(
-                request, lambda r: codec.decode_response(r),
-                lambda reg_id: map_remove_entry_listener_codec.encode_request(self.name, reg_id),
-                lambda m: codec.handle(m, self._handle_invalidation, self._handle_batch_invalidation))
-        except:
-            pass
+        codec = map_add_near_cache_invalidation_listener_codec
+        request = codec.encode_request(self.name, EntryEventType.INVALIDATION, self._is_smart)
+        self._invalidation_listener_id = self._register_listener(
+            request, lambda r: codec.decode_response(r),
+            lambda reg_id: map_remove_entry_listener_codec.encode_request(self.name, reg_id),
+            lambda m: codec.handle(m, self._handle_invalidation, self._handle_batch_invalidation)).result()
```

Comment:
Is it ok that `_add_near_cache_invalidation_listener` is still blocking? I didn't check how it's used further, so that's why I'm asking.

Reply:
This is called on the constructor of the near cached map if the invalidate on change option is true. I believe we should wait for this to complete. This is not a problem because `.get_XXX` methods on the Hazelcast client are blocking too. This was like this long time, so I didn't want to change that

Reply:
I see. That's weird that we don't have non-blocking `get_XXX` calls, but that weirdness is also present in Java client. Thanks for the clarification.